### ISSUE TITLE
__new__ should be a static method

### ISSUE THREAD

**Issue by Lynskylate** [RustPython/RustPython#1230]
In RustPython, \_\_new\_\_ is bound method but it should be a static method.

```python
class A:
    def __new__(*args, **kwargs):
        print("Args Total: {}".format(len(args)+len(kwargs)))

class B(A):
    def __new__(cls, *args, **kwargs):
        return super().__new__(cls, *args, **kwargs)
"""
>>> B()
CPython output
Args Total: 1 
RustPython output
Args Total: 2
"""
```
 super().__new__(cls, *args, **kwargs) will pass two same ype class(args[0] is same as cls).

[object \_\_new\_\_ Python docuemtion](https://docs.python.org/3/reference/datamodel.html#basic-customization)

**Comment by cthulahoops** [comment]
I think this is probably a duplicate of #778 . Just trying to get this stuff fresh in my mind so I can review the work here.

**Comment by cthulahoops** [comment]
I've been considering this a problem with ```super``` as it works in the examples without ```super``` but finally got a simple example that shows the difference without using ```super```:

CPython:

```
>>> (1).__new__(int, 5)
5
```

RustPython:

```
>>>>> (1).__new__(int, 5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Expected type <class 'type'>, not <class 'int'>
```

**Comment by Lynskylate** [comment]
I try to fix RustPython/RustPython#1241, but I met some trouble. I simplified the code in enum.py that caused the error. 
```python
class EnumMeta(type):
    def __new__(metacls, cls, bases, classdict):
        enum_class = super().__new__(metacls, cls, bases, classdict)
        return enum_class

class Test(metaclass=EnumMeta):
    def __init__(self):
        pass
"""
>>> t = Test() 
# RustPython Output:
Traceback (most recent call last):
  File "/tmp/reproduce.py", line 17, in <module>
    class Test(metaclass=EnumMeta):
  File "/tmp/reproduce.py", line 14, in __new__
    enum_class = super().__new__(metacls, cls, bases, classdict)
TypeError: type() takes 1 or 3 arguments,
"""
```

**Comment by Lynskylate** [comment]
Incompact example code for RustPython object's\ _\_new\_\_.
```python
type(object.__new__)
type(object().__new__)
```
RustPython:
```
<class 'builtin_function_or_method'>
<class 'method'>
```
CPython
```
<class 'builtin_function_or_method'>
<class 'builtin_function_or_method'>
```
